### PR TITLE
feat(hummock): introduce aligned allocator and cache file skeleton

### PR DESCRIPTION
**Pull request by MrCroxx**
## What's changed and what's your intention?

This is a one of the PRs for the file cache system, to avoid sending a huge PR all at once.

1. Introduce `AlignedAllocator` for direct I/O buffer allocation.
2. Introduce cache file skeleton.

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Refer to a related PR or issue link (optional)
#198

### CODE REVIEW

**Comment by MrCroxx** [comment]
Maybe this PR is too immature, but I want to keep each PR small and easy to review, QwQ.

File: ci/scripts/common.env.sh [hunk 1]
```diff
@@ -1,3 +1,4 @@
 export CARGO_TERM_COLOR=always
 export RUSTFLAGS="-D warnings"
 export PROTOC_NO_VENDOR=true
+export RISINGWAVE_CI=true
```
Comment (TennyZhuang) [inline ci/scripts/common.env.sh:4]: ?

Comment (MrCroxx) [reply ci/scripts/common.env.sh:4]: The unit test need to use ext4/xfs fs. With CI environment, I reuse the `/risingwave` directory to create a tempdir.

**Comment by TennyZhuang** [comment]
LGTM for the allocator. I haven't enough background knowledge to review the PR (

File: src/storage/src/hummock/file_cache/manager.rs [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    Xfs,
+}
+
+#[derive(Debug)]
+pub struct FileCacheManagerOptions {
+    pub dir: String,
+}
+
+pub struct FileCacheManager {
+    _dir: String,
```
Comment (TennyZhuang) [inline src/storage/src/hummock/file_cache/manager.rs:33]: It's not a good pattern in rust that an used variable is prefixed with underscore.

File: src/storage/src/hummock/file_cache/mod.rs [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+pub mod alloc;
+pub mod error;
+pub mod file;
+pub mod manager;
+
+async fn asyncify<F, T>(f: F) -> error::Result<T>
```
Comment (fuyufjh) [inline src/storage/src/hummock/file_cache/mod.rs:20]: Why not just call `spawn_blocking` directly? This seems opaque to me.

https://github.com/singularity-data/risingwave/blob/d3dc4dea4ca8cbc31db885045c642288c6211f73/src/storage/src/hummock/file_cache/file.rs#L128

File: src/storage/src/hummock/file_cache/manager.rs [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+            EXT4_SUPER_MAGIC => FsType::Ext4,
+            // FYI: https://github.com/nix-rust/nix/issues/1742
+            NixFsType(libc::XFS_SUPER_MAGIC) => FsType::Xfs,
+            nix_fs_type => return Err(Error::UnsupportedFilesystem(nix_fs_type.0)),
+        };
+        let fs_block_size = fs_stat.block_size() as usize;
+
+        Ok(Self {
+            _dir: options.dir,
+            _fs_type: fs_type,
```
Comment (fuyufjh) [inline src/storage/src/hummock/file_cache/manager.rs:57]: What will `fs_type` be used for?

**Review by fuyufjh — APPROVED** [review]
Generally LGTM. Since it's not completed yet, we may improve later.

File: src/storage/src/hummock/file_cache/manager.rs (hunk 4 above)
Comment (MrCroxx) [reply src/storage/src/hummock/file_cache/manager.rs:57]: I'm not sure if we need to support more file system with other pattern. I'll remove it if it's never used after the feature is little maturer. (Or remove now.

**Review by hzxa21 — APPROVED** [review]

**Pull request merged**